config/src/main/java/com/typesafe/config/ConfigRenderOptions.java

```diff
@@ -64,7 +69,7 @@ public ConfigRenderOptions setComments(boolean value) {
         if (value == comments)
             return this;
         else
-            return new ConfigRenderOptions(originComments, value, formatted, json);
+            return new ConfigRenderOptions(originComments, value, formatted, json, new DefaultComparator());
```

Comment:
These setters should be copying the existing comparator I believe. (otherwise setting anything other than the comparator would reset the comparator.) Meaning change `new DefaultComparator()` to `comparator` for all the `set*` methods.

Reply:
Fixed